src/main/java/uk/gov/ch/controller/update/OverseasEntityDataController.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.ch.controller.update;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.ch.exception.EntityEmailNotFoundException;
+import uk.gov.ch.exception.InvalidEntityNumberException;
+import uk.gov.ch.model.update.OverseasEntityDataJson;
+import uk.gov.ch.service.update.impl.OverseasOverseasEntityDataServiceImpl;
+import uk.gov.ch.util.ApiLogger;
+
+import javax.validation.ConstraintViolationException;
+import javax.validation.Valid;
+import javax.validation.constraints.Pattern;
+import java.util.HashMap;
+
+import static uk.gov.ch.util.Constants.OVERSEAS_ENTITY_ID_KEY;
+
+@RestController
+public class OverseasEntityDataController {
+    @Autowired
+    private OverseasOverseasEntityDataServiceImpl entityDataService;
+
+    @GetMapping("/overseas-entity/{incorporationNumber}/entityData")
+    public ResponseEntity<OverseasEntityDataJson> getEntityEmail(@PathVariable @Valid @Pattern(regexp = "^OE\\d{6}$") String incorporationNumber) {
```

Comment:
need a test for this validation in Karate

Reply:
I suggest creating a sub-task off a JIRA story for this and referencing it here, once created.

Reply:
https://companieshouse.atlassian.net/browse/UAR-477

Reply:
Thanks.